Makefile

```diff
@@ -0,0 +1,139 @@
+.PHONY: test cover gofmt gofmt-fix license-check clean tar.gz docker-push release docker-push-all
+
+# Registry used for publishing images
+REGISTRY?=quay.io/coreos
+
+# Default tag and architecture. Can be overridden
+TAG?=$(shell git describe --tags --dirty)
+ARCH?=amd64
+
+# These variables can be overridden by setting an environment variable.
+TEST_PACKAGES?=pkg/ip subnet remote
+TEST_PACKAGES_EXPANDED=$(TEST_PACKAGES:%=github.com/coreos/flannel/%)
+PACKAGES?=$(TEST_PACKAGES) network
+PACKAGES_EXPANDED=$(PACKAGES:%=github.com/coreos/flannel/%)
+
+# Set the (cross) compiler to use for different architectures
+ifeq ($(ARCH),amd64)
+	CC=gcc
+	IMAGE_SUFFIX=
```

Comment:
I'd prefer if you set `IMAGE_SUFFIX` to `-$(ARCH)` for all cases
And then do an `ifeq ($(ARCH),amd64)` and push without suffix as well for amd64